unstructured/ingest/connector/astra.py

```diff
@@ -123,3 +131,124 @@ def normalize_dict(self, element_dict: dict) -> dict:
             "content": element_dict.pop("text", None),
             "metadata": element_dict,
         }
+
+
+@dataclass
+class AstraIngestDoc(IngestDocCleanupMixin, BaseSingleIngestDoc):
+    connector_config: SimpleAstraConfig
+    metadata: t.Dict[str, str] = field(default_factory=dict)
+    registry_name: str = "astra"
+
+    @property
+    def filename(self):
+        return (
+            Path(self.read_config.download_dir)
+            / self.connector_config.collection_name
+            / f"{self.metadata['_id']}.txt"
+        ).resolve()
+ 
+    @property
+    def _output_filename(self):
+        return Path(self.processor_config.output_dir) / f"{self.filename}.json"
```

Comment:
This is getting saved in wrong directory due to self.filename being a path. Should be something like:
```
        return (Path(self.processor_config.output_dir)
            / self.connector_config.collection_name
            / f"{self.metadata['_id']}.json"
            )
```